lib/penthouse/tenants/octopus_schema_tenant.rb

```diff
@@ -21,30 +21,36 @@ class OctopusSchemaTenant < SchemaTenant
       # @yield [SchemaTenant] The current tenant instance
       # @return [void]
       def call(shard: Octopus.master_shard, &block)
-        Octopus.using(shard) do
-          super(&block)
-        end
+        switch_shard(shard: shard) { super(&block) }
       end
 
       # creates the tenant schema within the master shard
       # @see Penthouse::Tenants::SchemaTenant#create
       # @return [void]
       def create(**)
-        call { super }
+        switch_shard { super }
```

Comment:
Might be best to explicitly pass `shard: Octopus.master_shard` to these calls, e.g. `switch_shard(shard: Octopus.master_shard) { super }` that way it's explicit what we're doing.